internal/services/automanage/automanage_configuration_resource.go

```diff
@@ -0,0 +1,486 @@
+	DefenderForCloudEnabled   bool                       `tfschema:"defender_for_cloud_enabled"`
+	GuestConfigurationEnabled bool                       `tfschema:"guest_configuration_enabled"`
+	StatusChangeAlertEnabled  bool                       `tfschema:"status_change_alert_enabled"`
+
+	Location string            `tfschema:"location"`
+	Tags     map[string]string `tfschema:"tags"`
+}
+
+type AntimalwareConfiguration struct {
+	Enabled                   bool                    `tfschema:"enabled"`
+	Exclusions                []AntimalwareExclusions `tfschema:"exclusions"`
+	RealTimeProtectionEnabled bool                    `tfschema:"real_time_protection_enabled"`
+	ScheduledScanEnabled      bool                    `tfschema:"scheduled_scan_enabled"`
+	ScanType                  string                  `tfschema:"scheduled_scan_type"`
+	ScanDay                   int                     `tfschema:"scheduled_scan_day"`
+	ScanTimeInMinutes         int                     `tfschema:"scheduled_scan_time_in_minutes"`
+}
+
+type AntimalwareExclusions struct {
+	Extensions string `tfschema:"extensions"`
+	Paths      string `tfschema:"paths"`
+	Processes  string `tfschema:"processes"`
+}
+
+type AutoManageConfigurationResource struct{}
+
+var _ sdk.ResourceWithUpdate = AutoManageConfigurationResource{}
+
+func (r AutoManageConfigurationResource) ResourceType() string {
+	return "azurerm_automanage_configuration"
+}
+
+func (r AutoManageConfigurationResource) ModelObject() interface{} {
+	return &ConfigurationModel{}
+}
+
+func (r AutoManageConfigurationResource) IDValidationFunc() pluginsdk.SchemaValidateFunc {
+	return validate.AutomanageConfigurationID
+}
+
+func (r AutoManageConfigurationResource) Arguments() map[string]*pluginsdk.Schema {
+	return map[string]*pluginsdk.Schema{
+		"name": {
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			ForceNew:     true,
+			ValidateFunc: validation.StringIsNotEmpty,
+		},
+
+		"resource_group_name": commonschema.ResourceGroupName(),
+
+		"location": commonschema.Location(),
+
+		// "Antimalware/Enable": boolean,
+		// "Antimalware/EnableRealTimeProtection": boolean,
+		// "Antimalware/RunScheduledScan": boolean,
+		// "Antimalware/ScanType": string ("Quick", "Full"),
+		// "Antimalware/ScanDay": int (0-8) Ex: 0 - daily, 1 - Sunday, 2 - Monday, .... 7 - Saturday, 8 - Disabled,
+		// "Antimalware/ScanTimeInMinutes": int (0 - 1440),
+		// "Antimalware/Exclusions/Extensions": string (extensions separated by semicolon. Ex: ".ext1;.ext2"),
+		// "Antimalware/Exclusions/Paths": string (Paths separated by semicolon. Ex: "c:\excluded-path-1;c:\excluded-path-2"),
+		// "Antimalware/Exclusions/Processes": string (Processes separated by semicolon. Ex: "proc1.exe;proc2.exe"),
+		"antimalware": {
+			Type:     pluginsdk.TypeList,
+			Optional: true,
+			MaxItems: 1,
+			Elem: &pluginsdk.Resource{
+				Schema: map[string]*pluginsdk.Schema{
+					"enabled": {
```

Comment:
We shouldn't be exposing this field for blocks - the behaviour for features turned on/off should be if block is present -> feature is on, block is absent -> feature is off. We have a [section on this in the contributor docs](https://github.com/hashicorp/terraform-provider-azurerm/blob/main/contributing/topics/schema-design-considerations.md#features-that-are-toggled-by-the-property-enabled).

Reply:
Thanks for the review, update according to the guide. Removing the enable property.